lib/log/log.c

```diff
@@ -55,31 +55,34 @@ SPDK_LOG_REGISTER_TRACE_FLAG("debug", SPDK_TRACE_DEBUG)
 int
 spdk_set_log_facility(const char *facility)
 {
-	if (strcasecmp(facility, "daemon") == 0) {
-		spdk_g_log_facility = LOG_DAEMON;
-	} else if (strcasecmp(facility, "auth") == 0) {
-		spdk_g_log_facility = LOG_AUTH;
-	} else if (strcasecmp(facility, "authpriv") == 0) {
-		spdk_g_log_facility = LOG_AUTHPRIV;
-	} else if (strcasecmp(facility, "local1") == 0) {
-		spdk_g_log_facility = LOG_LOCAL1;
-	} else if (strcasecmp(facility, "local2") == 0) {
-		spdk_g_log_facility = LOG_LOCAL2;
-	} else if (strcasecmp(facility, "local3") == 0) {
-		spdk_g_log_facility = LOG_LOCAL3;
-	} else if (strcasecmp(facility, "local4") == 0) {
-		spdk_g_log_facility = LOG_LOCAL4;
-	} else if (strcasecmp(facility, "local5") == 0) {
-		spdk_g_log_facility = LOG_LOCAL5;
-	} else if (strcasecmp(facility, "local6") == 0) {
-		spdk_g_log_facility = LOG_LOCAL6;
-	} else if (strcasecmp(facility, "local7") == 0) {
-		spdk_g_log_facility = LOG_LOCAL7;
-	} else {
-		spdk_g_log_facility = LOG_DAEMON;
-		return -1;
+	int i;
+
+	for (i = 0; facilitynames[i].c_name != NULL; i++) {
+		if (strcasecmp(facilitynames[i].c_name, facility) == 0) {
+			spdk_g_log_facility = facilitynames[i].c_val;
+			return 0;
+		}
 	}
-	return 0;
+
+	spdk_g_log_facility = LOG_DAEMON;
+	return -1;
+}
+
+char *
+spdk_get_log_facility(void)
```

Comment:
spdk_get_log_facility() ought to return a const char * - at least on FreeBSD, the c_name field is a const char *, so returning it below causes a warning.